src/snapred/backend/service/NormalizationService.py

```diff
@@ -147,11 +149,62 @@ def normalization(self, request: NormalizationRequest):
             detectorPeaks=ingredients.detectorPeaks,
         ).dict()
 
+    def validateRequest(self, request: NormalizationRequest):
+        """
+        Validate the normalization request.
+
+        :param request: a normalization request
+        :type request: NormalizationRequest
+        """
+        if not self._sameStates(request.runNumber, request.backgroundRunNumber):
+            raise ValueError("Run number and background run number must be of the same Instrument State.")
+
+        # This is a redundant call, but it is placed here to facilitate re-sequencing.
+        permissionsRequest = CalibrationWritePermissionsRequest(
+            runNumber=request.runNumber, continueFlags=request.continueFlags
+        )
+        self.validateWritePermissions(permissionsRequest)
+
+    def validateWritePermissions(self, request: CalibrationWritePermissionsRequest):
+        """
+        Validate that the normalization-calibration workflow will be able to save its output.
+
+        :param request: a write-permissions request containing the run number and existing continue flags
+        :type request: CalibrationWritePermissionsRequest
+        """
+        # Note: this is split-out as a separate method so it can be checked as early as possible in the workflow.
+        continueFlags = ContinueWarning.Type.UNSET
+
+        # check that the user has write permissions to the save directory
+        if not self.checkWritePermissions(request.runNumber):
+            continueFlags |= ContinueWarning.Type.NO_WRITE_PERMISSIONS | ContinueWarning.Type.CONTINUE_NOT_POSSIBLE
```

Comment:
I do see the value in a `CONTINUE_NOT_POSSIBLE` for uncertain results failing from a continue anyway, but in this case would it not just be an exception?  Lack of write perms is fatal to the save calibration process.

Reply:
I had to think about this a bit. It seems to me that the only reason it's _fatal_ is that so much of the rest of SNAPRed depends on the precise location, format, and _name_ of the saved _calibration_ data files + metadata files.  What I'm trying to do here is allow the user to "muck around" in the panels and to see what's possible.  For example, maybe they _do_ want to do their own calibration, but these messages will mean that they know they need to talk to the CIS first, and / or might help them with _which_ paths they need to adjust in the `application.yml`.

Reply:
My argument for this living in the `ContinueWarning` is _twofold_.  (1) It makes a lot of sense in terms of implementation, as _many_ of these cases might be _changed_ in the future, in order to _allow_ a different level of continuation.  And (2) it also makes a kind-of logical sense, if you consider it as one end of the coninuum -- it's not "ridiculous" that it would belong there!

For example, (if I'm remembering correctly) I ran into case (1) _immediately_ in converting this PR to "staging".  On staging, there is _NO_ allowed continuation for the "no calibration" case, but on "next" there _is_ allowed continuation.